src/include/detail/linalg/tdb_matrix_with_ids.h

```diff
@@ -50,7 +50,7 @@
  */
 template <
     class T,
-    class IdsType = size_t,
+    class IdsType = uint64_t,
```

Comment:
Yep.  I think the rule of thumb should be -- don't use size_t for data -- only for indexing into data (which is how it is defined and intended to be used).  I have a bad habit of over-using it.